kod/object/item.kod

```diff
@@ -613,6 +613,52 @@ messages:
       return 1;
    }
 
+   GetItemRarityGrade(what = $)
+   "Returns the appropriate text coloring tier for the item."
+   {
+      local irarity, i;
+      irarity = ITEM_RARITY_GRADE_NORMAL;
+
+      % Provide default value if not an item
+      if NOT IsClass(self,&Item)
+      {
+         return ITEM_RARITY_GRADE_NORMAL;
+      }
+
+      if Send(self,@IsCursed)
+      {
+         return ITEM_RARITY_GRADE_CURSED;
+      }
+
+      % Static class rarities
+      if IsClass(self,&helm)               % Magic Spirit Helm
```

Comment:
This is not good object-oriented design.  We are up in "Item" here, which is an extremely high-level and general class.  There should be no mention of lower classes like Helm.

Reply:
Refactored as suggested.